tests/src/test/java/alluxio/client/cli/fsadmin/BackupCommandIntegrationTest.java

```diff
@@ -87,4 +101,54 @@ public void timeout() throws Exception {
       assertNotEquals(0, errCode);
     }
   }
+
+  @Test
+  public void workersLostRepro() throws Exception {
+    // Grab the master state-change lock via reflection.
+    MasterProcess masterProcess =
+        Whitebox.getInternalState(mLocalAlluxioCluster.getLocalAlluxioMaster(), "mMasterProcess");
+    MasterContext masterCtx = Whitebox.getInternalState(masterProcess, "mContext");
+    StateLockManager stateLockManager = masterCtx.getStateLockManager();
+
+    BlockMaster blockMaster = masterProcess.getMaster(BlockMaster.class);
+    Whitebox.setInternalState(blockMaster, "mMasterContext", masterCtx);
+
+    FileSystemMaster fsMaster = masterProcess.getMaster(FileSystemMaster.class);
+    Whitebox.setInternalState(fsMaster, "mMasterContext", masterCtx);
+
+    FileSystem fs = mLocalAlluxioCluster.getClient();
+    FileOutStream fsOutStr = fs.createFile(new AlluxioURI("/test"));
+    fsOutStr.write("bogus".getBytes());
+    // Lock the state-change lock on the master.
+    LockResource lr =
+        stateLockManager.lockExclusive(new StateLockOptions(false, false, true, 0, 0, 0, 100));
```

Comment:
@gpang if you pass the last argument of StateLockOptions as `0` (which means no interrupt-waiters cycle), then the repro will fail as before.